plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/kotlin/KSClassDeclarationImpl.kt

```diff
@@ -70,6 +71,9 @@ class KSClassDeclarationImpl private constructor(val ktClassOrObject: KtClassOrO
 
     override val primaryConstructor: KSFunctionDeclaration? by lazy {
         ktClassOrObject.primaryConstructor?.let { KSFunctionDeclarationImpl.getCached(it) }
+            ?: (ResolverImpl.instance.resolveDeclaration(ktClassOrObject) as? ClassDescriptor)?.unsubstitutedPrimaryConstructor?.let {
+                KSFunctionDeclarationDescriptorImpl.getCached(it)
```

Comment:
Just realized this will result in the primary constructor to have an origin of Orgin.Class, which is not reflecting the fact. Might be better to have something like Origin.Synthetic.

Reply:
I guess that declared primary constructors don't include init blocks while the synthetic ones do. Is that true?

Reply:
I was not intended to address init block with this PR, but that's a good point and I can check that.